phalcon/db/adapter/pdo.zep

```diff
@@ -85,9 +85,12 @@ abstract class Pdo extends Adapter
 	 */
 	public function connect(descriptor = null)
 	{
-		var username, password, dsnParts, dsnAttributes,
+		var eventsManager,
+			username, password, dsnParts, dsnAttributes,
 			persistent, options, key, value;
 
+		let eventsManager = <ManagerInterface> this->_eventsManager;
```

Comment:
The problem is not having a new events manager it is rather not being able to attach to it early enough.  So this doesn't help.

```php
    if typeof eventsManager != "object" {
        let eventsManager = new \Phalcon\Events\Manager();
        let this->_eventsManager = eventsManager;
    }
```

This approach would help but then it seems to remove flexibility from a project that uses more than one DB service of different setups and even DB types since a plugin would need to attach to all global `db` events.  In this case the user would need to check for stuff in the plugin that it shouldn't know about and it would be bad design.

```php
    if typeof eventsManager != "object" {
        var di = \Phalcon\Di::getDefault();
        let eventsManager = di->getShared('eventsManager');
        let this->_eventsManager = eventsManager;
    }
```